### DISCUSSION TITLE
Strange behavior with Time axis series

### DISCUSSION THREAD

**Discussion by logikdev** [GSI-CS-CO/chart-fx#471] [General]
Hello! 

I am trying to display a chart containing a few time series and had initially got it to work by using simple incrementing integers for the X axis, which resulted in this:

![initial_chart](https://user-images.githubusercontent.com/6424131/142477969-b8620de9-41e0-41b3-95b8-5e9eff5eb4d8.png)

Since i want to perform FFTs on these datasets, i needed to convert the X axis to be a time axis. I used the TimeAxis sample as a guide. This worked as expected for the FFT charts, but had an odd side effect as you can see here: 

![artefacts](https://user-images.githubusercontent.com/6424131/142478363-34649b7f-7587-4620-9212-1b7d383f5e6a.png)

The following code block produces the initial result:

``` java
 dataSetPressureL.add(i, datapoint.pressureLValue); 
 dataSetPressureR.add(i, datapoint.pressureRValue); 
 dataSetSum.add(i, (datapoint.pressureLValue + datapoint.pressureRValue));
 dataSetW.add(i, review_powerHashMap.get(datapoint.now.getTime() / 1000));
 dataSetHR.add(i, review_heartrateHashMap.get(datapoint.now.getTime() / 1000));
 dataSetCad.add(i, review_cadenceHashMap.get(datapoint.now.getTime() / 1000));
```

While this code block produces the strange artefacts:

``` java
 dataSetPressureL.add((datapoint.now.getTime() + tz.getOffset(datapoint.now.getTime())) / 1000, datapoint.pressureLValue); 
 dataSetPressureR.add((datapoint.now.getTime() + tz.getOffset(datapoint.now.getTime())) / 1000, datapoint.pressureRValue);
 dataSetSum.add((datapoint.now.getTime() + tz.getOffset(datapoint.now.getTime())) / 1000, (datapoint.pressureLValue + datapoint.pressureRValue));
 dataSetW.add((datapoint.now.getTime() + tz.getOffset(datapoint.now.getTime())) / 1000, review_powerHashMap.get(datapoint.now.getTime() / 1000));
 dataSetHR.add((datapoint.now.getTime() + tz.getOffset(datapoint.now.getTime())) / 1000, review_heartrateHashMap.get(datapoint.now.getTime() / 1000));
 dataSetCad.add((datapoint.now.getTime() + tz.getOffset(datapoint.now.getTime())) / 1000, review_cadenceHashMap.get(datapoint.now.getTime() / 1000));
```

I'm not quite sure how to attack this and would appreciate any suggestion or help anyone would provide. Thank you!

**Comment by wirew0rm** [comment]
Hey and welcome to chart-fx :wave: 

If your `datapoint.now.getTime()` function returns an integer, your division by `1000` will be performed as an integer division, so you will only get points at full seconds, which would explain how your plot is looking. Try to change this to `/ 1000.0` (or even better `* 1e-3`) and you should be good to go. If this is not the problem, please let me know so I can look into it.

**Reply by logikdev** [reply]
Hi! 

Thanks for the code block edits and the prompt reply! Of course, as i was writing this post, it got my mind thinking about things a bit differently and checked the individual timestamps i was feeding to the datasets and you are 100% correct in your assessment. 

I will implement your suggestions and go from there. Since this isn't strictly a Chartfx issue, is it worthwhile to leave this discussion up for others to find or should i remove it entirely?

**Reply by wirew0rm** [reply]
Glad that it works now :)
As this is a somewhat common mistake, I think others can also benefit from it, too.
I personally also like these questions, as they are our main way to see what people are doing with the library.
So no need to remove it from my POV.

**Reply by logikdev** [reply]
It's not fully functional yet but your help put me on the right path to address the calculations i was doing wrong in regards to integer/double precision so that's what i'm working on at the moment. 

Thanks again for your help. This is a great project. I've very to have found it while researching for my own project. Cheers!
